packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx

```diff
@@ -1,14 +1,23 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import styles from './verify_account.module.css';
 import { FaRegWindowMinimize } from 'react-icons/fa';
 import { useLocation } from 'react-router-dom';
 import axios from 'axios';
 import { verifyURL } from '../../../../Global';
 
 export const VerifyAccount = ({ className, onSuccess }) => {
+    const [jwtToken, setJwtToken] = useState('');
     const location = useLocation();
-    const params = new URLSearchParams(location.search);
-    const jwtToken = params.get('jwt_token');
+    const [buttonState, setButtonState] = useState(true);
+
+    useEffect(() => {
+        const params = new URLSearchParams(location.search);
+        if (params.size === 0) {
+            window.location.href = '/hackaubg';
+        } else {
+            setJwtToken(params.get('jwt_token'));
+        }
+    }, [location]);
```

Comment:
I would usually suggest extracting those as hooks but at this point just make it work.